app/queries/keyword_search_entries_query.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+class KeywordSearchEntriesQuery
+  def initialize(relation = KeywordSearchEntry)
+    @relation = relation
+  end
+
+  def with_keyword(keyword_id)
+    @relation = relation.where(keyword_id: keyword_id)
+
+    self
+  end
+
+  def top_ads_count
+    relation.where(kind: :ads, position: :top).size
```

Comment:
How about using the `COUNT in SQL` directly, if yes, let change for another methods as well 👍 

```suggestion
    relation.count(kind: :ads, position: :top)
```

Reply:
We could bypass using `count` if the associated records are already loaded. See the difference between `length`, `count`, and `size` 👉 https://stackoverflow.com/a/6083229

Reply:
Ah, thanks @olivierobert then `size` will be the right solution here 👍